src/core-services/product/utils/applyProductFilters.js

```diff
@@ -0,0 +1,190 @@
+import SimpleSchema from "simpl-schema";
+
+const filters = new SimpleSchema({
+  "productIds": {
+    type: Array,
+    optional: true
+  },
+  "productIds.$": String,
+  "shopIds": {
+    type: Array,
+    optional: true
+  },
+  "shopIds.$": String,
+  "tagIds": {
+    type: Array,
+    optional: true
+  },
+  "tagIds.$": String,
+  "query": {
+    type: String,
+    optional: true
+  },
+  "isArchived": {
+    type: Boolean,
+    optional: true
+  },
+  "isVisible": {
+    type: Boolean,
+    optional: true
+  },
+  "metafieldKey": {
+    type: String,
+    optional: true
+  },
+  "metafieldValue": {
+    type: String,
+    optional: true
+  },
+  "priceMin": {
+    type: Number,
+    optional: true
+  },
+  "priceMax": {
+    type: Number,
+    optional: true
+  }
+});
+
+/**
+ * @name applyProductFilters
+ * @summary Builds a selector for Products collection, given a set of filters
+ * @private
+ * @param {Object} context - an object containing the per-request state
+ * @param {Object} productFilters - See filters schema above
+ * @returns {Object} Selector
+ */
+export default function applyProductFilters(context, productFilters) {
+  // if there are filter/params that don't match the schema
+  filters.validate(productFilters);
+
+  // Init default selector - Everyone can see products that fit this selector
+  let selector = {
+    ancestors: [], // Lookup top-level products
```

Comment:
Nothing for you to do, just making a note for myself to possibly update this in the auth updates with the new style